src/Hosting/Hosting/src/Internal/HostingLoggerExtensions.cs

```diff
@@ -11,28 +11,18 @@
 
 namespace Microsoft.AspNetCore.Hosting
 {
-    internal static class HostingLoggerExtensions
+    internal static partial class HostingLoggerExtensions
     {
-        private static readonly Action<ILogger, string, Exception?> _startupAssemblyLoaded =
-                LoggerMessage.Define<string>(LogLevel.Debug, LoggerEventIds.HostingStartupAssemblyLoaded, "Loaded hosting startup assembly {assemblyName}", skipEnabledCheck: true);
-
-        private static readonly Action<ILogger, string, Exception?> _listeningOnAddress =
-                LoggerMessage.Define<string>(LogLevel.Information, LoggerEventIds.ServerListeningOnAddresses, "Now listening on: {address}");
-
         public static IDisposable RequestScope(this ILogger logger, HttpContext httpContext)
         {
             return logger.BeginScope(new HostingLogScope(httpContext));
         }
 
-        public static void ListeningOnAddress(this ILogger logger, string address)
-        {
-            _listeningOnAddress(logger, address, null);
-        }
+        [LoggerMessage(EventId = LoggerEventIds.ServerListeningOnAddresses, EventName = "ServerListeningOnAddresses", Level = LogLevel.Information, Message = "Now listening on: {address}")]
```

Comment:
We want to be explicit about the event name. It is easy for someone to unknowingly change the event name by renaming a method. That would introduce a breaking change.

That is the reason why we have strings today instead of using `nameof(MethodName)`.